openpype/hosts/houdini/plugins/publish/collect_karma_rop.py

```diff
@@ -0,0 +1,136 @@
+        if parm.path() in processed:
+            raise RuntimeError("Parameter references result in cycle.")
+
+        processed.add(parm.path())
+
+        ref = parm.getReferencedParm()
+        if ref.path() == parm.path():
+            # It returns itself when it doesn't reference
+            # another parameter
+            return ref
+        else:
+            parm = ref
+
+
+def evalParmNoFrame(node, parm, pad_character="#"):
+
+    parameter = node.parm(parm)
+    assert parameter, "Parameter does not exist: %s.%s" % (node, parm)
+
+    # If the parameter has a parameter reference, then get that
+    # parameter instead as otherwise `unexpandedString()` fails.
+    parameter = get_top_referenced_parm(parameter)
+
+    # Substitute out the frame numbering with padded characters
+    try:
+        raw = parameter.unexpandedString()
+    except hou.Error as exc:
+        print("Failed: %s" % parameter)
+        raise RuntimeError(exc)
+
+    def replace(match):
+        padding = 1
+        n = match.group(2)
+        if n and int(n):
+            padding = int(n)
+        return pad_character * padding
+
+    expression = re.sub(r"(\$F([0-9]*))", replace, raw)
+
+    with hou.ScriptEvalContext(parameter):
+        return hou.expandStringAtFrame(expression, 0)
+
+
+class CollectKarmaROPRenderProducts(pyblish.api.InstancePlugin):
+    """Collect Karma Render Products
+
+    Collects the instance.data["files"] for the multipart render product.
+
+    Provides:
+        instance    -> files
+
+    """
+
+    label = "Karma ROP Render Products"
+    order = pyblish.api.CollectorOrder + 0.4
+    hosts = ["houdini"]
+    families = ["karma_rop"]
+
+    def process(self, instance):
+
+        rop = hou.node(instance.data.get("instance_node"))
+
+        # Collect chunkSize
+        chunk_size_parm = rop.parm("chunkSize")
+        if chunk_size_parm:
+            chunk_size = int(chunk_size_parm.eval())
+            instance.data["chunkSize"] = chunk_size
+            self.log.debug("Chunk Size: %s" % chunk_size)
+
+            default_prefix = evalParmNoFrame(rop, "picture")
+            render_products = []
+
+            # Default beauty AOV
+            beauty_product = self.get_render_product_name(
+                prefix=default_prefix, suffix=None
+            )
+            render_products.append(beauty_product)
+
+            files_by_aov = {
+                "beauty": self.generate_expected_files(instance,
+                                                       beauty_product)
+            }
+
+            filenames = list(render_products)
+            instance.data["files"] = filenames
+
+        for product in render_products:
+            self.log.debug("Found render product: %s" % product)
+
+        if "expectedFiles" not in instance.data:
+            instance.data["expectedFiles"] = list()
+        instance.data["expectedFiles"].append(files_by_aov)
+
+    def get_render_product_name(self, prefix, suffix):
+        if suffix:
```

Comment:
This could also be done such as:
```py
product_name = prefix
 
if suffix:
    # Add ".{suffix}" before the extension
    prefix_base, ext = os.path.splitext(prefix)
    product_name = "{0}.{1}{2}".format(prefix_base, suffix, ext)

```